### ISSUE TITLE
uuid.UUID should be considered immutable by default

### ISSUE THREAD

**Issue by sk-** [astral-sh/ruff#16750]
### Summary

`uuid.UUID` is part of the standard library and should be treated as an immutable call by default. Without it some rules report errors, like RUF009

```
sample.py:32:22: RUF009 Do not perform function call `uuid.UUID` in dataclass defaults
   |
32 |     baz: uuid.UUID = uuid.UUID("12345678-1234-5678-1234-567812345678")
   |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ RUF009
```

I know this could be "solved" by manually adding `"uuid.UUID"` to the setting `lint.flake8-bugbear.extend-immutable-calls`, but given this is a symbol from the standard library it'd make sense to be there by default.

### Version

ruff 0.10.0 (27e9d1fe3 2025-03-13)

**Comment by MichaReiser** [comment]
Thanks. We should also consider adding the other `UUID` factory methods

**Comment by CarliJoy** [comment]
I would be nice if there is a user configurable list of "immutable" functions.
I.e. `attr.Factory` is very valid but found as false positive.
So is [`typed_settings.option`](https://typed-settings.readthedocs.io/en/latest/apiref.html#typed_settings.option) and a number of other candidates.

Allowing a user definable list for valid function for RUF009, removes work from the ruff project because projects can remove false-positives themself.
